perfscale_regression_ci/scripts/common.sh

```diff
@@ -138,4 +138,94 @@ function uncordon_all_nodes() {
   for worker in ${worker_nodes}; do
     oc adm uncordon $worker
   done
+}
+
+# pass $namespace $deployment_name $initial_pod_num $final_pod_num
+# e.g. delete_project "test=concurent-job"
+function check_deployment_pod_scale()
+{
+	namespace=$1
+	deployment_name=$2
+	initial_pod_num=$3
+	final_pod_num=$4
+
+	# Sometimes, it takes a while for pods to scale (up or down). Decrement the counter each time we chack for the pods (in 
+	# the deployment). The pods should scale before count_scaling reaches a negative value, but if it does become negative, 
+	# give an error message and exit the test. This same logic follows for count_running. It takes some time for the pods to 
+	# terminate (scale down) or start running (scale up). The pods should all be in a Running state before count_running 
+	# reaches a negative value. If count_running ecome negative, give an error message and exit the test.
+	count_scaling=1000
+	count_running=1000
+
+	while [[ ( $initial_pod_num -ne $final_pod_num ) && ( count_scaling -gt 0 ) ]];
+	do 
+		oc get deployment $deployment_name  -n $namespace
+		initial_pod_num=$(oc get deployment $deployment_name --no-headers -n $namespace | awk -F ' {1,}' '{print $4}' )
+		((count_scaling--))
+	done
+
+	if [ $count_scaling -lt 0 ]; then
+		echo "pods did not not scale to $final_pod_num"
+		exit 1
+	fi
+
+  pods_not_running=$(oc get pods -n $namespace --no-headers | egrep -v "Completed|Running" | wc -l)
+	echo "pods not running (due to scaling): $pods_not_running"
+
+	while [[ ( $count_running -gt 0 ) && ( $pods_not_running -gt 0 ) ]];
+	do
+		pods_not_running=$(oc get pods -n $namespace --no-headers | egrep -v "Completed|Running" | wc -l)
```

Comment:
do you think we are able to use _wait_for_obj_creation_ function here instead?

Reply:
In this case, I'm trying to get the exact time the pods become ready. In wait_for_obj_creation, there is a 5 second sleep. This would extra time to the result.